platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,272 @@
+ * crlLocationUrl}.
+ */
+public class CrlGenerator implements Runnable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CrlGenerator.class);
+
+  private static final PropertiesLoader PROPERTIES_LOADER = PropertiesLoader.getInstance();
+
+  private static final String HTTPS = "https://";
+
+  private static final int INITIAL_DELAY = 0;
+
+  private static final int SCHEDULER_INTERVAL = 30;
+
+  private static final int FILE_BACKED_STREAM_THRESHOLD = 1000000;
+
+  static final String CRL_PROPERTY_KEY = Merlin.OLD_PREFIX + Merlin.X509_CRL_FILE;
+
+  static final String DEM_CRL = "/localCrl.crl";
+
+  static final String PEM_CRL = "/localCrl.pem";
+
+  @VisibleForTesting
+  static String issuerEncryptionPropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/issuer/encryption.properties").getPath();
+
+  @VisibleForTesting
+  static String issuerSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/issuer/signature.properties").getPath();
+
+  @VisibleForTesting
+  static String serverEncryptionPropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/encryption.properties").getPath();
+
+  @VisibleForTesting
+  static String serverSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/signature.properties").getPath();
+
+  @VisibleForTesting
+  static String crlFileLocation = new AbsolutePathResolver("etc/keystores").getPath();
+
+  private final ClientFactoryFactory factory;
+
+  private final ScheduledExecutorService scheduler;
+
+  private String crlLocationUrl;
+
+  private boolean crlByUrlEnabled;
+
+  private ScheduledFuture<?> handle;
+
+  public CrlGenerator(ClientFactoryFactory factory) {
+    this.factory = factory;
+    this.scheduler =
+        Executors.newSingleThreadScheduledExecutor(
+            StandardThreadFactoryBuilder.newThreadFactory("crlThread"));
+  }
+
+  /** Pulls down the CRL file provided via URL and writes it locally. */
+  @Override
+  public synchronized void run() {
+
+    if (!crlByUrlEnabled || crlLocationUrl == null || !crlLocationUrl.startsWith(HTTPS)) {
+      return;
+    }
+
+    FileBackedOutputStream fileBackedOutputStream = null;
+
+    try {
+      SecureCxfClientFactory cxfClientFactory =
+          factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+      WebClient client = cxfClientFactory.getWebClient();
+      Response response = client.get();
+
+      Object entity = response.getEntity();
+      if (entity instanceof InputStream) {
+
+        // Read the response content and get the byte source
+        ByteSource byteSource;
+        try (InputStream entityStream = (InputStream) response.getEntity()) {
+          fileBackedOutputStream = new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
+          IOUtils.copy(entityStream, fileBackedOutputStream);
+          byteSource = fileBackedOutputStream.asByteSource();
```

Comment:
✏️ Reverse these 2 lines in case they ever change their implementation. First closing the stream before getting something from it is safer